extended ci flows

andreas-kupries opened the pull request:
Ref #2100 

 - 3a - variant of 3 (RKE, private CA, external registry) where nginx is default ingress controller :heavy_check_mark: 
 - 7 - See 3a, with S3 storage moved to external, AWS S3 bucket. :heavy_check_mark: 
 - 8 - See 7, metallb removed - :boom: 

metallb is an outer lb for dns to point at (magic dns/domain with ip from the rke cluster) - now trialing change to use route53, directly pointing at the nginx IC (see s4), obviating the need

andreas-kupries committed:
scenario3 variant  using nginx as default ingress controller
i.e. rke + private ca + external registry + nginx ingress

scenario 3 variant - nginx as default ingress, s3 storage external on AWS
called scenario7, as it needed changes in the go code

fixups

fixup: disable traefik references and checks in the new variants

fixup: 3a scenario addressing

yav (yet another variant) with metallb dropped from the setup

same tracking fixup in the new flows. separate for easier future squash

enrichman reviewed (changes requested):
When we install/upgrade during tests we need to add also the `server.disableTracking=true` flag. If it's not present also in the other scenarios would you mind adding it?

```
"--set", "server.disableTracking=true", // disable tracking during tests
```

Also I would drop the commented tests if those are just leftovers or copy-paste from the other ones.

andreas-kupries commented:
There will be a general cleanup of commented parts, likely also a reduction of variants. The various variants is me incrementally working towards the proper test case in small steps.

Will add the `disable tracking`.

andreas-kupries committed:
s8 fails without the metallb
 - looks that the base s3 uses a semi-random ip for the cluster with magic dns for domain.
   the mlb is used to serve that ip as an outer lb routing to the inner IC (traefik, nginx, ...)
 - most other cases use route53 for dns instead, and have the domain pointed directly at the IC, by its ip or hostname

reworking s8 here to do the same. code is taken from s4/eks, as it also had the checks that nginx is up as IC, the equivalent of the removed traefik IC check from the s3 base.

andreas-kupries committed:
move trial into existing thing

andreas-kupries committed:
drop the old working variants
merge the latest variant without metalb and new route53 into the last working
i.e. single variant left, as s7, under trial

andreas-kupries committed:
fixup: typo, scenario name

andreas-kupries committed:
fixup: bump time to wait on nginx ingress to appear

andreas-kupries committed:
fixup: added missing init of zone id
shuffled code blocks a bit, match s4 order a bit more

andreas-kupries committed:
debug nginx bring up

andreas-kupries committed:
continue debug

andreas-kupries committed:
continued debug

andreas-kupries committed:
debug continue - log fix?

andreas-kupries committed:
logs may fail initially

andreas-kupries commented:
Originating issue was solved as issue in the home lab setup.
